File: app/src/main/java/com/coderstory/purify/utils/SharedHelper.java
package com.coderstory.purify.utils;

import android.content.Context;

import com.zlm.libs.preferences.PreferencesProviderUtils;

public class SharedHelper {
    String spName = "config";
    private Context context;

    public SharedHelper(Context context) {
        this.context = context;
    }

    public void put(String key, Object object) {

        if (object instanceof String) {
            PreferencesProviderUtils.putString(context, spName, key, (String) object);
        } else if (object instanceof Integer) {
            PreferencesProviderUtils.putInt(context, spName, key, (Integer) object);
        } else if (object instanceof Boolean) {
            PreferencesProviderUtils.putBoolean(context, spName, key, (Boolean) object);
        } else if (object instanceof Float) {
            PreferencesProviderUtils.putFloat(context, spName, key, (Float) object);
        } else if (object instanceof Long) {
            PreferencesProviderUtils.putLong(context, spName, key, (Long) object);
        } else {
            PreferencesProviderUtils.putString(context, spName, key, object.toString());
        }
    }


    public boolean getBoolean(String key, boolean defaultObject) {
        return PreferencesProviderUtils.getBoolean(context, spName, key, defaultObject);
    }

    public String getString(String key, String defaultObject) {
        return PreferencesProviderUtils.getString(context, spName, key, defaultObject);
    }
}

File: app/src/main/java/com/coderstory/flyme10/tools/SharedHelper.kt
package com.coderstory.flyme10.tools

import android.content.Context
import com.coderstory.flyme10.preferences.PreferencesProviderUtils

class SharedHelper(private val context: Context) {
    var spName = "UserSettings"
    fun put(key: String, value: Any): Boolean {
        return when (value) {
            is String -> {
                PreferencesProviderUtils.putString(context, spName, key, value)
            }

            is Int -> {
                PreferencesProviderUtils.putInt(context, spName, key, value)
            }

            is Boolean -> {
                PreferencesProviderUtils.putBoolean(context, spName, key, value)
            }

            is Float -> {
                PreferencesProviderUtils.putFloat(context, spName, key, value)
            }

            is Long -> {
                PreferencesProviderUtils.putLong(context, spName, key, value)
            }

            else -> {
                PreferencesProviderUtils.putString(context, spName, key, value.toString())
            }
        }
    }

    fun getBoolean(key: String, defaultObject: Boolean): Boolean {
        return PreferencesProviderUtils.getBoolean(context, spName, key, defaultObject)
    }

    fun getString(key: String, defaultObject: String): String {
        return PreferencesProviderUtils.getString(context, spName, key, defaultObject)
    }
}